sql/expression/logic.go

```diff
@@ -37,11 +37,19 @@ func JoinAnd(exprs ...sql.Expression) sql.Expression {
 	case 0:
 		return nil
 	case 1:
+		if exprs[0] == nil {
+			return nil
+		}
 		return exprs[0]
 	default:
+		if exprs[0] == nil {
+			return JoinAnd(exprs[1:]...)
+		}
```

Comment:
Was there a concrete case where `exprs` had `nil` expression values in it? Seems like if element 0 is nil, then there could be other nil elements in the slice, and since I wasn't expecting nil expressions to be passed in, I'm wondering if it's better to keep this simple and not try to silently drop any nil values.

Reply:
I adjusted the code to instead check for nil in NewAnd and NewOr which is a lot cleaner. I think it's still worthwhile to check for nil values if the original code checked for them (which it appears to have)